### ISSUE TITLE
Suggestion for syntax sugar for regression transform

### ISSUE THREAD

**Issue by jan-scorl** [altair-viz/altair#1968]
**tl;dr**
Can we make it possible to draw the regression line for a specific aggregation method by using:
```
chart.transform_regression(on='season', regression='max(value)').mark_line(color='red')
```

This is a direct reply to my issue (#1967), which I closed, because I was able to solve it myself.

Let's assume that I have a data series that I want to plot in two ways. 
1. I want to see the max value for each season.
2. I want to see the number of data points (count) for each season.
Doing so with Altair is quite simple.

```
max_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='max(value):Q',
)

count_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='count(value):Q',
)

(max_chart) | (count_chart)
```
![visualization-7](https://user-images.githubusercontent.com/17554280/74496454-b357ba80-4f15-11ea-8de8-d72986dc500a.png)

However, if I want to add a regression line for each chart, things didn't work as I expected them to work. I expected that either approach one or two would work.

**Approach 1**
I type `regression='value'` and Altair figures out the aggregation method by itself. 
```
max_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='max(value):Q',
)

count_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='count(value):Q',
)

max_regression = max_chart.transform_regression(on='season', regression='value').mark_line(color='red')
count_regression = count_chart.transform_regression(on='season', regression='value').mark_line(color='red')

(max_chart + max_regression) | (count_chart + count_regression)
```
![visualization-8](https://user-images.githubusercontent.com/17554280/74496495-c8cce480-4f15-11ea-8feb-aa3631b66475.png)

**Approach 2**
I type `regression='aggregation_method(value)'`
```
max_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='max(value):Q',
)

count_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='count(value):Q',
)

max_regression = max_chart.transform_regression(on='season', regression='max(value)').mark_line(color='red')
count_regression = count_chart.transform_regression(on='season', regression='count(value)').mark_line(color='red')

(max_chart + max_regression) | (count_chart + count_regression)
```
![visualization-9](https://user-images.githubusercontent.com/17554280/74496517-d84c2d80-4f15-11ea-8409-0208db124fb5.png)

Instead, I had to do this:
**Approach 3:**
```
max_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='value_max:Q',
).transform_aggregate(
    value_max='max(value)',
    groupby=['season']
)

count_chart = alt.Chart(bvb).mark_bar().encode(
    x='season:O',
    y='value_count:Q',
).transform_aggregate(
    value_count='count(value)',
    groupby=['season']
)

max_regression = max_chart.transform_regression(on='season', regression='value_max').mark_line(color='red')
count_regression = count_chart.transform_regression(on='season', regression='value_count').mark_line(color='red')


(max_chart + max_regression) | (count_chart + count_regression )
```
![visualization-10](https://user-images.githubusercontent.com/17554280/74496543-eb5efd80-4f15-11ea-9ef0-5703e2705e70.png)

I think it would be great if approach 2 would work, since it is in line with the syntax shortcuts when plotting a normal graph.

**Comment by jakevdp** [comment]
In order to make approach 2 work, the ``transform_regression`` method would have to add both an aggregate transform and a regression transform to the chart, which might be somewhat confusing. The reason we support aggregate shorthands within encodings is because in the Vega-Lite grammar, aggregates can appear in encodings. That is not the case within the regression transform grammar.
